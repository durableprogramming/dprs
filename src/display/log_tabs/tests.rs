
use insta::assert_snapshot;
use ratatui::{backend::TestBackend, Terminal};

use super::*;

#[test]
fn test_log_tabs_render() {
    let backend = TestBackend::new(100, 3);
    let mut terminal = Terminal::new(backend).unwrap();
    
    let titles = vec!["container1".to_string(), "container2".to_string(), "container3".to_string()];
    let log_tabs = LogTabs::new(titles);
    
    terminal
        .draw(|f| {
            render_log_tabs::<TestBackend>(f, &log_tabs, f.area());
        })
        .unwrap();
    
    assert_snapshot!(terminal.backend().to_string());
}

#[test]
fn test_log_tabs_with_selection() {
    let backend = TestBackend::new(100, 3);
    let mut terminal = Terminal::new(backend).unwrap();
    
    let titles = vec!["container1".to_string(), "container2".to_string(), "container3".to_string()];
    let mut log_tabs = LogTabs::new(titles);
    log_tabs.set_index(1); // Select second tab
    
    terminal
        .draw(|f| {
            render_log_tabs::<TestBackend>(f, &log_tabs, f.area());
        })
        .unwrap();
    
    assert_snapshot!(terminal.backend().to_string());
}

#[test]
fn test_log_tabs_empty() {
    let backend = TestBackend::new(100, 3);
    let mut terminal = Terminal::new(backend).unwrap();
    
    let log_tabs = LogTabs::new(vec![]);
    
    terminal
        .draw(|f| {
            render_log_tabs::<TestBackend>(f, &log_tabs, f.area());
        })
        .unwrap();
    
    assert_snapshot!(terminal.backend().to_string());
}

#[test]
fn test_log_tabs_styles() {
    let backend = TestBackend::new(100, 3);
    let mut terminal = Terminal::new(backend).unwrap();
    
    let titles = vec!["container1".to_string(), "container2".to_string(), "container3".to_string()];
    let log_tabs = LogTabs::new(titles);
    
    terminal
        .draw(|f| {
            render_log_tabs::<TestBackend>(f, &log_tabs, f.area());
        })
        .unwrap();
    
    // Verify output contains expected content
    let output = terminal.backend().to_string();
    assert!(output.contains("container1"));
    assert!(output.contains("container2"));
    assert!(output.contains("container3"));
    assert!(output.contains("Containers"));
}
