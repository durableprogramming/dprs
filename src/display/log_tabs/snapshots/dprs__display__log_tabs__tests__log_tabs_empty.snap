---
source: src/display/log_tabs/tests.rs
expression: terminal.backend().to_string()
---
"╭Containers────────────────────────────────────────────────────────────────────────────────────────╮"
"│                                                                                                  │"
"╰──────────────────────────────────────────────────────────────────────────────────────────────────╯"
